Assets/Scripts/EnemyBehaviour.cs

```diff
@@ -0,0 +1,197 @@
+/// \file Framework classes for Enemy actions and implementations thereof
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Enemy's level of involvement with the player
+/// </summary>
+public enum ENEMY_STATE
+{
+    /// <summary>
+    /// Enemy is unaware of player
+    /// </summary>
+    idle = 0,
+    /// <summary>
+    /// Enemy is aware of player but not engaged in conflict
+    /// </summary>
+    aware = 1,
+    /// <summary>
+    /// Enemy is engaged in conflict with player
+    /// </summary>
+    engaged = 2
+}
+
+/// <summary>
+/// Exception thrown in the event that an ENEMY_STATE value is incompatible with a function
+/// </summary>
+public class InvalidEnemyStateException : System.ArgumentException
+{
+    /// <summary>
+    /// Throws an exception with a given error message.
+    /// </summary>
+    /// <param name="message"> Error message </param>
+    public InvalidEnemyStateException(string message) : base(message)
+    {
+    }
```

Comment:
For empty bodies, I prefer the following style:
```suggestion
    public InvalidEnemyStateException(string message) : base(message) {}
```

What do you think?

Reply:
Yeah actually, I prefer having it on one line like you just edited it.

Would you be okay if I put a space between the curly braces? The advantage of that is just that vscode automatically puts a space there if you use its "format document" feature, which I've been using lately just because it makes it easier to adhere to a more consistent curly brace style.

Reply:
Sure

Reply:
Awesome. In that case, I just pushed this as a change in a recent update, so I'll resolve this for now since I've indirectly accepted the suggestion.